web/api/v1/api.go

```diff
@@ -1016,6 +1066,32 @@ func (api *API) respondError(w http.ResponseWriter, apiErr *apiError, data inter
 	}
 }
 
+func (api *API) respondWarnings(w http.ResponseWriter, warnings []error, data interface{}) {
+	seen := make(map[string]struct{}, len(warnings))
+	var warningStrings []string
+	for _, warning := range warnings {
+		// Deduplicate warnings
```

Comment:
This loses information, I'd just concatenate them.

This shouldn't need a new function.

Reply:
Why would this lose information? It's just removing duplicate warnings. If you query several metrics, each one is going to return a warning about the remote read issue, but all for the same source and same warning.

Reply:
What if two different remote reads produce the same error?

Reply:
Is that possible? The errors I get back are like `error sending request: Post http://localhost:9999/api/v1/read: dial tcp 127.0.0.1:9999: connect: connection refused`. If it were a different remote read wouldn't the ip/port have to be different too?

Reply:
Not necessarily, there's a few more options involved that could be in play.

Reply:
Ok, I assumed they'd have to be from the same source.

Reply:
In a typical setup yes, but there's more esoteric possibilities.